src_assets/common/assets/web/index.html

```diff
@@ -110,49 +107,41 @@ <h3>{{githubVersion.name}}</h3>
-        // If we can't get versions, return false
-        if (!this.githubVersion || !this.version) return false;
-
-        // Get the GitHub version tag
+        if (!this.githubVersion || !this.version) {
+          return false;
+        }
         let v_github = this.githubVersion.name;
-        // If the version starts with a v, remove it
-        if (v_github.indexOf("v") === 0) v_github = v_github.substring(1);
-
-        // Get the current version
+        if (v_github.indexOf("v") === 0) {
+          v_github = v_github.substring(1);
+        }
         let v_this = this.version;
-        // If the version starts with a v, remove it
-        if (v_this.indexOf("v") === 0) v_this = v_this.substring(1);
-
-        // if nightly or dirty, we do an additional check to make sure it's an actual upgrade.
-        if (this.buildVersionIsNightly || this.buildVersionIsDirty) {
-          const stableVersion = this.version.split('.').slice(0, 3).join('.');
-          return this.githubVersion.tag_name.substring(1) > stableVersion;
+        if (v_this.indexOf("v") === 0) {
+          v_this = v_this.substring(1);
         }
-
-        // return true if the version is different, otherwise false
         return v_github !== v_this;
       },
-      nightlyBuildAvailable() {
-        // Verify nightly data is available and the build version is not stable
-        // This is important to ensure the UI does not try to load undefined values.
-        if (!this.nightlyData || this.buildVersionIsStable) return false;
-        // If built with dirty git tree, return false
-        if (this.buildVersionIsDirty) return false;
-        // Get the commit hash
-        let commit = this.version?.split(".").pop();
-        // return true if the commit hash is different, otherwise false
-        return this.nightlyData.head_sha.indexOf(commit) !== 0;
+      preReleaseBuildAvailable() {
+        if (!this.githubPreRelease || !this.githubVersion || !this.version) {
+          return false;
+        }
+        let v_github_pre = this.githubPreRelease.name;
+        if (v_github_pre.indexOf("v") === 0) {
+          v_github_pre = v_github_pre.substring(1);
+        }
+        let v_github_stable = this.githubVersion.name;
+        if (v_github_stable.indexOf("v") === 0) {
+          v_github_stable = v_github_stable.substring(1);
+        }
+        let v_this = this.version;
+        if (v_this.indexOf("v") === 0) {
+          v_this = v_this.substring(1);
+        }
+        return v_github_pre !== v_this && new Date(this.githubPreRelease.published_at) > new Date(this.githubVersion.published_at);
       },
       buildVersionIsDirty() {
         return this.version?.split(".").length === 5 &&
           this.version.indexOf("dirty") !== -1
       },
-      buildVersionIsNightly() {
-        return this.version?.split(".").length === 4
-      },
-      buildVersionIsStable() {
-        return this.version?.split(".").length === 3
-      },
```

Comment:
There will no longer be a reliable way to determine if the user is using a pre-release or stable, so we probably need to add a config option to allow ignoring notifications for pre-releases.

Reply:
Can't we add a new flag to the build that just compiles "stable", "beta" or "dirty"? Given which pipeline, or lack thereof it ran?